olive/data/container/data_container.py

```diff
@@ -65,6 +65,18 @@ def create_dataloader(self):
         pre_process_dataset = self.pre_process(dataset)
         return self.dataloader(pre_process_dataset)
 
+    def create_snpe_dataloader(self, model, evaluation=True):
```

Comment:
>create_snpe_dataloader [](http://example.com/codeflow?start=8&length=22)

do we need add new signature?

Reply:
I just discussed this with Jiapeng. The snpe dataloader is used only for snpe model evaluation or quantization, so it is better kept as a separate method. Unless, we want to add different condition to the `create_dataloader` and `create_calibration_dataloader` methods.

Reply:
Is it possible there would be lot of customized create_xx_dataloader in future? that's my concerns.

Reply:
That is a valid concern. I added this directly to the container since `SNPEModel` is an olive model type and it also needs snpe dataloader. I think for other passes, they would need to do the conversion from olive dataloader to their specific type in the pass itself. 

I could also do the same. remove this method and instead get olive dataloader. And the convert it to snpe dataloader in the snpe evaluation and quantization pass.

Reply:
How about this? @jambayk 
You can inherit the RawDataContainer and change the dataloader to snpe_dataloader. In this way, we can keep the base class as simple as possible. Then the usage in SNPE case will be clearer.
```python
@Registry.register(DataContainerType.DATA_CONTAINER)
class SNPEDataContainer(RawDataContainer):
default_components_type: ClassVar[dict] = {
    DataComponentType.LOAD_DATASET.value: "raw_dataset"
    DataComponentType.DATALOADER.value: "default_snpe_dataloader"
}
```

Reply:
Or we can create snpe template in `https://github.com/microsoft/Olive/pull/333/files/f9b7c7eac227bfa2b9f6f05ff80516184b472c33#diff-bfa09a24344a0657b60de3f87a8b40be86b65f9e34ab3ac259f0c631ba9d6c3c` based on RawDataContainer.

Reply:
A bit differences with offline discussion where I were thinking to replace the default dataloader to `default_snpe_dataloader` in `RawDataContainer`. Then the `RawDataContainer` only serves SNPE. I agreed that is not the way to go.

[1] The new suggestion is that we keep the `RawDataContainer` and to produce the `SNPEDataContainer`.  Then in SNPE config, we can directly use `SNPEDataContainer`. But for onnx/torch model, we can use `RawDataContainer`. **That is determined by users.**

[2] For this, it sounds good to me as well. 
> If people are worried about the new create_snpe_dataloader method, I think a better solution is to not add the method. The evaluation and quantization code will then do the following:
datacontainer.create_dataloader -> regular dataloader -> Convert regular dataloader to snpe dataloader


For me, I have no strong tendency on method [1] or [2]. Both of them can resolve Mike's concern.

Reply:
Besides, quick question about this: I do not get your points. Is there any necessary case where we must call snpe_dataloader independently? From my understanding, all the method in data components can be call independently.
>  The snpe dataloader code has no dependency on the RawDataContainer. It can work with any datacontainer.

Reply:
actually, pushed the changes for [2] directly. Please let me know if that addresses the concerns.